src/Standards/Generic/Sniffs/CodeAnalysis/EmptyPHPStatementSniff.php

```diff
@@ -53,12 +53,25 @@ public function process(File $phpcsFile, $stackPtr)
         case 'T_SEMICOLON':
             $prevNonEmpty = $phpcsFile->findPrevious(Tokens::$emptyTokens, ($stackPtr - 1), null, true);
 
-            if ($prevNonEmpty === false
-                || ($tokens[$prevNonEmpty]['code'] !== T_SEMICOLON
+            if ($prevNonEmpty === false) {
+                return;
+            }
+
+            if ($tokens[$prevNonEmpty]['code'] !== T_SEMICOLON
                 && $tokens[$prevNonEmpty]['code'] !== T_OPEN_TAG
-                && $tokens[$prevNonEmpty]['code'] !== T_OPEN_TAG_WITH_ECHO)
+                && $tokens[$prevNonEmpty]['code'] !== T_OPEN_TAG_WITH_ECHO
             ) {
-                return;
+                if ($tokens[$prevNonEmpty]['code'] !== T_CLOSE_CURLY_BRACKET
+                    || isset($tokens[$prevNonEmpty]['scope_condition']) === false
+                ) {
+                    return;
+                }
+
+                $scopeOwner = $tokens[$tokens[$prevNonEmpty]['scope_condition']]['code'];
+                if ($scopeOwner === T_CLOSURE || $scopeOwner === T_ANON_CLASS) {
+                    return;
+                }
+                // Else, it's something like `if (foo) {};` and the semi-colon is not needed.
```

Comment:
According to the PHPCS native CS standard, you need a blank line above this line (after the control structure). Once that's in place, the build should pass.

Reply:
Thanks, it's done